openedx/core/djangoapps/agreements/api.py

```diff
@@ -68,3 +73,179 @@ def get_integrity_signatures_for_course(course_id):
+    """
+    Determines if a user needs to acknowledge the LTI PII Agreement
+
+    Arguments:
+        * username (str)
+
+    Returns:
+        * True if the user needs to sign a new acknowledgement.
+        * False if the acknowledgements are up to date.
+    """
+    course_has_lti_pii_tools = _course_has_lti_pii_tools(course_id)
+    signature_exists = _user_lti_pii_signature_exists(username, course_id)
+    signature_out_of_date = _user_signature_out_of_date(username, course_id)
+
+    if (course_has_lti_pii_tools and (not signature_exists) or 
+        (course_has_lti_pii_tools and signature_exists and signature_out_of_date)):
+        # write a new signature, or update an existing signature
+        return True
+    else:
+        # course does not have lti pii tools, or signatue is up to date
+        return False
+
+
+def _course_has_lti_pii_tools(course_id):
+    """
+    Determines if a specifc course has lti tools sharing pii
+
+    Arguments:
+        * course_id (str)
+
+    Returns:
+        * True if the course does have a list.
+        * False if the course does not.
+    """
+    course_key = CourseKey.from_string(course_id)
+    try:
+        LTIPIITool.objects.get(course_key=course_key)
+    except LTIPIITool.DoesNotExist:
+        # no entry in the database
+        return False
+    else:
+        return True
+
+
+def _user_lti_pii_signature_exists(username, course_id):
+    """
+    Determines if a user's lti pii signature exists for a specfic course
+
+    Arguments:
+        * username (str)
+        * course_id (str)
+
+    Returns:
+        * True if user has a signature for the given course.
+        * False if the user does not have a signature for the given course.
+    """
+    course_key = CourseKey.from_string(course_id)
+
+    try:
+        user = User.objects.get(username=username)
+        LTIPIISignature.objects.get(user=user, course_key=course_key)
+    except (User.DoesNotExist, LTIPIISignature.DoesNotExist):
+        return False
+    else:
+        return True  # signature exist
+
+
+def _user_signature_out_of_date(username, course_id):
+    """
+        Determines if a user's existing lti pii signature is out-of-date for a given course.
+
+        Arguments:
+            * username (str)
+            * course_id (str)
+
+        Returns:
+            * True if user needs a new signature
+            * False if the user has an up-to-date signature.
+        """
+    course_key = CourseKey.from_string(course_id)
+
+    try:
+        user = User.objects.get(username=username)
+        user_lti_pii_signature_hash = LTIPIISignature.objects.get(course_key=course_key, user=user).lti_tools_hash
+        course_lti_pii_tools_hash = LTIPIITool.objects.get(course_key=course_key).lti_tools_hash
+    except (User.DoesNotExist, LTIPIISignature.DoesNotExist, LTIPIITool.DoesNotExist):
+        return False
+    else:
+        if user_lti_pii_signature_hash == course_lti_pii_tools_hash:
```

Comment:
Nit. You can shorten this a bit, if you want. Because `a==b` evaluates to a boolean itself, you can just return the result of your logical expression. For example, `return user_lti_pii_signature_hash != course_lti_pii_tools_hash` will accomplish the exact same thing and is a little easier to read.